code/graphics/software/FontManager.cpp

```diff
@@ -310,8 +310,117 @@ namespace font
 		}
 	}
 
+	// This function will extract the font family name (Name ID 1) from TrueType font data.
+	// It handles UCS-2 (UTF-16BE) to UTF-8 conversion.
+	static SCP_string extractFamilyNameFromTTF(const ubyte* data, int size)
+	{
+		// TTF/OTF fonts start the table directory at byte 12.
+		int table_offset = 12;
```

Comment:
I know this is not likely to be touched again, but this should be a `const` or  `constexpr` as it should never be changed.